Library/Homebrew/utils/backtrace.rb

```diff
@@ -0,0 +1,25 @@
+# typed: true
+# frozen_string_literal: true
+
+module Utils
+  module Backtrace
+    # Cleans `sorbet-runtime` gem paths from the backtrace unless...
+    # 1. `verbose` is set
+    # 2. first backtrace line starts with `sorbet-runtime`
+    #   - This implies that the error is related to Sorbet.
+    sig { params(error: Exception).returns(T.nilable(T::Array[String])) }
+    def self.clean(error)
+      backtrace = error.backtrace
+
+      return backtrace if Context.current.verbose?
+      return backtrace if backtrace.blank?
+      return backtrace if backtrace.fetch(0).start_with?(sorbet_runtime_path)
+
+      backtrace.reject { |line| line.start_with?(sorbet_runtime_path) }
```

Comment:
I think it would be nice to print a one-liner something like:
```ruby
if any_lines_removed
  opoo "Removed Sorbet lines from backtrace!"
  unless Homebrew::EnvConfig.no_env_hints?
    puts "Rerun with --verbose or HOMEBREW_SORBET_BACKTRACES=1 to see these lines"
  end
end
```

obviously adjusting messaging based on what we decide to go with.

Otherwise I can't see how you'd know really to run `--verbose` and also it feels much more useful to tell people when the lines were removed.

Reply:
Alternatively: add these lines to the start of the backtrace instead.

Reply:
I'm still not a fan of adding another environment variable (I think that can wait until multiple people complain about it) but adding a message makes sense. I guess we'd only need to print it the first time a backtrace is modified, right? Does that sound reasonable?

Reply:
Then it would look like this:

```console
$ brew fetch homebrew/cask/calhash --os=mojave -d
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Formula/calhash.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading homebrew/cask/calhash
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading calhash
Error: attempted to use a Downloadable without a URL!
Warning: Removed Sorbet lines from backtrace!
Rerun with --verbose to see the original backtrace
/usr/local/Homebrew/Library/Homebrew/downloadable.rb:75:in `downloader'
/usr/local/Homebrew/Library/Homebrew/downloadable.rb:50:in `cached_download'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:236:in `fetch_fetchable'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:201:in `fetch_cask'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:178:in `block (3 levels) in fetch'
/usr/local/Homebrew/Library/Homebrew/simulate_system.rb:29:in `with'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:166:in `block (2 levels) in fetch'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:163:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:163:in `block in fetch'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:94:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/fetch.rb:94:in `fetch'
/usr/local/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'
```